tools/tests/com.liferay.ide.project.core.tests/src/com/liferay/ide/project/core/tests/modules/ServiceCommandTests.java

```diff
@@ -20,65 +20,301 @@
 import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertTrue;
 
-import org.eclipse.core.runtime.IStatus;
-
-import aQute.remote.api.Agent;
+import java.io.File;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLConnection;
 
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.FileLocator;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.debug.core.DebugPlugin;
+import org.eclipse.debug.core.ILaunchConfigurationType;
+import org.eclipse.debug.core.ILaunchConfigurationWorkingCopy;
+import org.eclipse.debug.core.ILaunchManager;
+import org.eclipse.wst.server.core.IRuntimeWorkingCopy;
+import org.eclipse.wst.server.core.IServer;
+import org.eclipse.wst.server.core.IServerWorkingCopy;
+import org.eclipse.wst.server.core.ServerCore;
+import org.eclipse.wst.server.core.model.ServerDelegate;
 import org.junit.Test;
 
+import com.liferay.ide.core.util.FileUtil;
+import com.liferay.ide.core.util.LaunchHelper;
+import com.liferay.ide.project.core.ProjectCore;
 import com.liferay.ide.project.core.modules.ServiceCommand;
-import com.liferay.ide.server.util.SocketUtil;
+import com.liferay.ide.server.core.LiferayServerCore;
+import com.liferay.ide.server.core.portal.PortalRuntime;
+import com.liferay.ide.server.core.tests.ServerCoreBase;
 
 /**
  * @author Lovett Li
  */
-public class ServiceCommandTests
+public class ServiceCommandTests extends ServerCoreBase
 {
+    private PortalRuntime portalRuntime = null;
 
-    @Test
-    public void GetIntegrationPoints() throws Exception
+    public void startServer() throws Exception
     {
-        if( canConnect() )
+        final IServer server = getServer();
+
+        final String exceFileName = Platform.getOS().contains( "win" ) ? "catalina.bat" : "catalina.sh";
+
+        final LaunchHelper launchHelper = new LaunchHelper();
+
+        launchHelper.setLaunchSync( false );
+
+        final IPath serverLocation = server.getRuntime().getLocation().append( "bin" );
+
+        setupAgent();
+
+        launchHelper.launch( getLaunchConfig( serverLocation, exceFileName, "run" ), ILaunchManager.RUN_MODE, null );
```

Comment:
We already have code in other parts of server core test that launch the server.  Can those not be used instead?  See REmoteServerManagerTests